driver-core/src/main/java/com/datastax/driver/core/policies/RackAwarePolicy.java

```diff
@@ -0,0 +1,330 @@
+    this.usedHostsPerRemoteDc = usedHostsPerRemoteDc;
+    this.usedHostsPerRemoteRack = usedHostsPerRemoteRack;
+  }
+
+  @Override
+  public void init(Cluster cluster, Collection<Host> hosts) {
+    if (!localDc.equals(UNSET) && !localRack.equals(UNSET))
+      logger.info(
+          "Using provided data-center name '{}' and rack name '{}' for RackAwarePolicy",
+          localDc,
+          localRack);
+
+    for (Host host : hosts) {
+      String dc = dc(host);
+      String rack = dc + "-" + rack(host); // Add DC as prefix to rack name to make it unique
+
+      if (localDc.equals(UNSET) && !dc.equals(UNSET)) {
+        logger.info(
+            "Using provided data-center name '{}' for RackAwarePolicy (if this is incorrect, please provide the correct datacenter name with RackAwarePolicy constructor)",
+            dc);
+        localDc = dc;
+      }
+
+      if (localRack.equals(UNSET) && !rack(host).equals(UNSET)) {
+        logger.info(
+            "Using provided rack name '{}' for RackAwarePolicy (if this is incorrect, please provide the correct rack name with RackAwarePolicy constructor)",
+            rack(host));
+        localRack = rack;
+      }
+
+      CopyOnWriteArrayList<Host> currentDcLiveHosts = perDcLiveHosts.get(dc);
+      if (currentDcLiveHosts == null) {
+        perDcLiveHosts.put(dc, new CopyOnWriteArrayList<Host>(Collections.singletonList(host)));
+      } else {
+        currentDcLiveHosts.addIfAbsent(host);
+      }
+
+      CopyOnWriteArrayList<Host> currentRackLiveHosts = perRackLiveHosts.get(rack);
+      if (currentRackLiveHosts == null) {
+        perRackLiveHosts.put(rack, new CopyOnWriteArrayList<Host>(Collections.singletonList(host)));
+      } else {
+        currentRackLiveHosts.addIfAbsent(host);
+      }
+    }
+
+    this.index.set(new Random().nextInt(Math.max(hosts.size(), 1)));
+  }
+
+  private String dc(Host host) {
+    String dc = host.getDatacenter();
+    return dc == null ? localDc : dc;
+  }
+
+  private String rack(Host host) {
+    String rack = host.getRack();
+    return rack == null ? localRack : rack;
+  }
+
+  /**
+   * Return the HostDistance for the provided host.
+   *
+   * <p>This policy considers nodes in the local datacenter and local rack as {@code LOCAL}. For
+   * each remote rack in the local datacenter, it considers a configurable number of hosts as {@code
+   * REMOTE} and the rest is {@code IGNORED}. For each remote datacenter it considers a configurable
+   * number of hosts as {@code FOREIGN} and the rest is {@code IGNORED}.
+   *
+   * @param host the host of which to return the distance of.
+   * @return the HostDistance to {@code host}.
+   */
+  @Override
+  public HostDistance distance(Host host) {
+    String dc = dc(host);
+    String rack = dc + "-" + rack(host);
+
+    if (localDc.equals(UNSET)
+        || localRack.equals(UNSET)
+        || dc.equals(localDc) && rack(host).equals(localRack)) {
```

Comment:
The condition represents this logic: if `localDc` is `UNSET` then the node is considered local, otherwise if `localRack` is `UNSET` then it is again considered local, otherwise host's dc and rack should match to `localDc` and `localRack` so that it is considered local.

Reply:
I do not think if the host's dc is `UNSET` it should be considered as local.

Reply:
So is the logic in `DCAwareRoundRobinPolicy` a bug?

Reply:
Actually, I tried to understand the logic but could not yet find some relevant commits about it.

Reply:
Your explanation seems sound. Can you open an issue about logic in DCAwareRoundRobinPolicy? I'll try to look into it in the future

Reply:
I forgot to mention the opened issue here: https://github.com/scylladb/java-driver/issues/223